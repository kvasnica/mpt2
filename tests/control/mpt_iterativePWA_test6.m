function mpt_iterativePWA_test1

% prior to 2.0.4 there was a bug which caused mpt_iterativePWA to break with
% error if user-defined target set was given, but no dynamics contained the
% origin in it's interior.

sysStruct.A = { eye(2), eye(2) };
sysStruct.B = { eye(2), eye(2) };
sysStruct.C = { eye(2), eye(2) };
sysStruct.D = { eye(2), eye(2) };
sysStruct.umax = ones(2,1);
sysStruct.umin = -ones(2,1);
sysStruct.ymax = ones(2,1);
sysStruct.ymin = -ones(2,1);

% define dynamics such that they do not contain the origin
H = [eye(2); -eye(2)];
K = [3; 1; -1; 1];
sysStruct.guardX = { H, H };
sysStruct.guardC = { K, K };

probStruct.subopt_lev = 1;
probStruct.N = Inf;
probStruct.Q = eye(2);
probStruct.R = eye(2);
probStruct.norm = 2;

% define custom Tset
probStruct.Tset = unitbox(2,1);

% this test should NOT break
ctrl = mpt_control(sysStruct, probStruct, struct('maxiterations', 1));
