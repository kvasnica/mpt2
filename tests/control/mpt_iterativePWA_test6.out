WARNING: Terminal set provided, you should set "probStruct.Tconstraint=2" to enforce it...


Using User-Defined Target Set...
Warning: this may not guarantee stability
Warning: setting probStruct.P_N to stage cost.

dynamics 1 overlaps with dynamics 2 in the X space
dynamics 2 overlaps with dynamics 1 in the X space
mpt_iterativePWA: union of terminal sets is convex, merging...
Generating feasible set
new horizon: 1        
Maximum number of iterations reached without convergence...
Please increase the value of Options.maxiterations and run the procedure again.
Maximum number of iterations reached without convergence...
Please increase the value of Options.maxiterations and run the procedure again.
1 hulls to be explored
Generating regions
dynamics 2, hull 1/1        

1 regions generated
Removing unused regions (backwards)...
dynamics 2, hull 1/1           


Solution consists of 1 regions
